iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/ShortBackupRetentionCheck.java

```diff
@@ -69,7 +90,75 @@ private void checkBackupRetentionDatabaseAccounts(ContextualResource resource) {
     }
   }
 
-  private static Predicate<Expression> isNumericValue(Predicate<Float> predicate) {
+  private void checkBackupRetentionRecoveryServicesVaults(ContextualResource resource) {
+    retrieveRetentionPolicyObjects(resource).forEach(retentionPolicyObject -> {
+      ContextualObject retentionDurationObject = retrieveRetentionDurationObject(retentionPolicyObject);
+      Double durationInDays = computeRetentionInDays(retentionDurationObject);
+      if (durationInDays != null && durationInDays < retentionPeriodInDays) {
+        retentionDurationObject.property("count").report(RETENTION_PERIOD_TOO_SHORT_MESSAGE, retentionDurationObject.property("durationType").toSecondary("Duration type"));
+      }
+    });
+  }
+
+  private static Stream<ContextualObject> retrieveRetentionPolicyObjects(ContextualResource resource) {
+    ContextualProperty backupManagementType = resource.property("backupManagementType");
+
+    if (backupManagementType.is(isValue(TYPES_BASIC_RETENTION::contains))) {
+      return Stream.of(resource.object("retentionPolicy"));
+    } else if (backupManagementType.is(isValue(TYPES_SUBPROTECTION_RETENTION::contains))) {
+      return resource.list("subProtectionPolicy")
+        .objects()
+        .map(contextualObject -> contextualObject.object("retentionPolicy"));
+    } else {
+      return Stream.empty();
+    }
+  }
+
+  private static ContextualObject retrieveRetentionDurationObject(ContextualObject retentionPolicyObject) {
+    ContextualProperty retentionPolicyType = retentionPolicyObject.property("retentionPolicyType");
+
+    if (retentionPolicyType.is(isValue("SimpleRetentionPolicy"::equals))) {
```

Comment:
You are using `isValue(String::eqals)` quite often. Maybe it's time for an `isEqual(String)` predicate?